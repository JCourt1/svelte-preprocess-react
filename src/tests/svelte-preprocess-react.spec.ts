import { describe, expect, it } from "vitest";
import { preprocess } from "svelte/compiler";
import preprocessReact from "../lib/svelte-preprocess-react";
import source from "./fixtures/Container.svelte?raw";

describe("preprocess-react", () => {
  it("should process <react:component> tags", async () => {
    const output = await preprocess(source, preprocessReact(), {
      filename: "Counter.svelte",
    });
    expect(output.code).toMatchSnapshot();
  });
});
