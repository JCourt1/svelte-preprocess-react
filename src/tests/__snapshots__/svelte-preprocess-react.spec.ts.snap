// Vitest Snapshot v1

exports[`preprocess-react > should process <react:component> tags 1`] = `
"<script>import sveltifyReact from \\"svelte-preprocess-react/sveltifyReact18\\";
  // @ts-nocheck
  let count = 1;
const React$Counter = sveltifyReact(Counter);</script>

<React$Counter
  {count}
  onCount={(next) => {
    count = next;
  }}
/>
"
`;
